src/Test/Router/RequestRouterTest.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace CF\Router\Test;
+
+use CF\Integration\DefaultIntegration;
+use CF\Router\RequestRouter;
+
+class RequestRouterTest extends \PHPUnit_Framework_TestCase
+{
+    protected $mockConfig;
+    protected $mockAPI;
+    protected $mockIntegration;
+    protected $mockDataStore;
+    protected $mockLogger;
+    protected $mockRequest;
+    protected $requestRouter;
+
+    public function setup()
+    {
+        $this->mockConfig = $this->getMockBuilder('CF\Integration\DefaultConfig')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->mockAPI = $this->getMockBuilder('CF\Integration\IntegrationAPIInterface')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->mockDataStore = $this->getMockBuilder('CF\Integration\DataStoreInterface')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->mockLogger = $this->getMockBuilder('CF\Integration\DefaultLogger')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->mockIntegration = new DefaultIntegration($this->mockConfig, $this->mockAPI, $this->mockDataStore, $this->mockLogger);
+
+        $this->mockRequest = $this->getMockBuilder('CF\API\Request')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->requestRouter = new RequestRouter($this->mockIntegration);
+    }
+
+    public function testAddRouterAddsRouter() {
+        $this->requestRouter->addRouter('CF\API\Client', null);
+        $this->assertEquals('CF\Router\DefaultRestAPIRouter', get_class($this->requestRouter->getRouterList()[0]));
```

Comment:
```
public function addRouter($clientClassName, $routes) {
        $client = new $clientClassName($this->integrationContext);
        $router = new DefaultRestAPIRouter($this->integrationContext, $client, $routes);
        array_push($this->routerList, $router);
    }
```
just realized we can add the same class multiple times to the router where it shouldn't be allowed.